test/components/account/login.spec.js

```diff
@@ -233,4 +243,175 @@ describe('AccountLogin', () => {
+      const a = component.get('a');
+      a.element.addEventListener('click', (event) => {
+        event.preventDefault();
+      });
+      await a.trigger('click');
+      a.classes('disabled').should.be.true();
+    });
+
+    it('aborts navigation after the link is clicked', async () => {
+      const app = await load('/login', { container: oidcContainer })
+        .restoreSession(false);
+      const a = app.get('#account-login a');
+      a.element.addEventListener('click', (event) => {
+        event.preventDefault();
+      });
+      await a.trigger('click');
+      const result = await app.vm.$router.push('/not-found');
+      isNavigationFailure(result, NavigationFailureType.aborted).should.be.true();
+    });
+  });
+
+  describe('next query parameter if OIDC is enabled', () => {
+    it('appends ?next to the link', async () => {
+      const component = await load('/login?next=%2Fusers', {
+        container: oidcContainer,
+        root: false
+      });
+      const href = component.get('a').attributes('href');
+      href.should.equal('/v1/oidc/login?next=%2Fusers');
+    });
+
+    it('does not append ?next if it was specified twice', async () => {
+      const component = await load('/login?next=%2Fusers&next=%2Faccount%2Fedit', {
+        container: oidcContainer,
+        root: false
+      });
+      component.get('a').attributes('href').should.equal('/v1/oidc/login');
+    });
+
+    it('does not append ?next if it has no value', async () => {
+      const component = await load('/login?next', {
+        container: oidcContainer,
+        root: false
+      });
+      component.get('a').attributes('href').should.equal('/v1/oidc/login');
+    });
+  });
+
+  describe('OIDC error', () => {
+    it('shows an alert for auth-ok-user-not-found', async () => {
+      const component = await load('/login?oidcError=auth-ok-user-not-found', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.alert('danger', (message) => {
+        message.should.startWith('There is no Central account associated with your email address.');
+      });
+    });
+
+    it('shows an alert for email-claim-not-provided', async () => {
+      const component = await load('/login?oidcError=email-claim-not-provided', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.alert('danger', (message) => {
+        message.should.startWith('Central could not access the email address associated with your account.');
+      });
+    });
+
+    it('shows an alert for email-not-verified', async () => {
+      const component = await load('/login?oidcError=email-not-verified', {
+        container: oidcContainer,
+        root: false
+      });
+      component.should.alert('danger', (message) => {
+        message.should.startWith('Your email address has not been verified by your login server.');
+      });
+    });
+
+    it('does not show an alert if there are two errors', async () => {
```

Comment:
That's not gonna happen the way backend works but I think we should give first error message the priority

Reply:
If Backend doesn't work that way, I think we can just ignore it: it would be extra code to use either error message. This test is mostly here just to verify that nothing breaks if the query parameter is specified twice. I thought about skipping the test entirely — like you said, it'd be pretty strange for that to happen!